javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -1277,6 +1277,9 @@ private static MethodNamer resolveMethodNamer(ProxyMethod proxyMethod, Convenien
         if (!isProtocolMethod && convenienceApi != null) {
             return new MethodNamer(convenienceApi.getName());
         } else {
+            if (proxyMethod.isSync()) {
+                return new MethodNamer(proxyMethod.getName().replace("Sync", ""));
```

Comment:
It would be safer to only remove the trailing "Sync" from name (or call the `getBaseName`)?

As it might affect the API name if it contains "Sync" e.g. "getSynchronizedResult" (for now just an idea, not a real name, but I guess we would eventually encounter one having "Sync" in it)